examples/basics/gloo/spatial_filters.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# vispy: gallery 2
+"""
+Example demonstrating spatial filtering using spatial-filters fragment shader.
+Left and Right Arrow Keys toggle through available filters.
+"""
+
+import os
+import numpy as np
+
+from vispy.util.transforms import ortho
+from vispy.io import _data_dir
+from vispy import gloo
+from vispy import app
+
+
+# Image to be displayed
+W, H = 100, 100
+I = np.random.uniform(0,1,(5,5,3)).astype(np.float32)
+
+# A simple texture quad
+data = np.zeros(4, dtype=[('a_position', np.float32, 2),
+                          ('a_texcoord', np.float32, 2)])
+data['a_position'] = np.array([[0, 0], [W, 0], [0, H], [W, H]])
+data['a_texcoord'] = np.array([[0, 0], [0, 1], [1, 0], [1, 1]])
+
+
+VERT_SHADER = """
+// Uniforms
+uniform mat4 u_model;
+uniform mat4 u_view;
+uniform mat4 u_projection;
+uniform float u_antialias;
+
+// Attributes
+attribute vec2 a_position;
+attribute vec2 a_texcoord;
+
+// Varyings
+varying vec2 v_texcoord;
+
+// Main
+void main (void)
+{
+    v_texcoord = a_texcoord;
+    gl_Position = u_projection * u_view * u_model * vec4(a_position,0.0,1.0);
+}
+"""
+
+FRAG_SHADER = """
+#include "misc/spatial-filters.frag"
+uniform sampler2D u_texture;
+uniform vec2      u_shape;
+uniform float     u_interpolation;
+varying vec2 v_texcoord;
+void main()
+{
+    //gl_FragColor = texture2D(u_texture, v_texcoord);
+    //gl_FragColor.a = 1.0;
+    
```

Comment:
Here you can add a comment saying those tests are currently inefficient even it is not really the point here.